src/lambda/wizards/samInitWizard.ts

```diff
@@ -80,6 +97,199 @@ export class DefaultCreateNewSamAppWizardContext extends WizardContext implement
         return val ? (val.label as Runtime) : undefined
     }
 
+    public async promptUserForTemplate(
+        currRuntime: Runtime,
+        currTemplate?: SamTemplate
+    ): Promise<SamTemplate | undefined> {
+        const templates = getSamTemplateWizardOption(currRuntime)
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.template.prompt', 'Select a SAM Application Template'),
+                value: currTemplate
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: templates.toArray().map(template => ({
+                label: template,
+                alwaysShow: template === currTemplate,
+                description:
+                    template === currTemplate
+                        ? localize('AWS.wizard.selectedPreviously', 'Selected Previously')
+                        : getTemplateDescription(template)
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        //eventBridgeStarterAppTemplate requires aws credentials
+        if (val && val.label === eventBridgeStarterAppTemplate) {
+            if (!this.currentCredentials) {
+                vscode.window.showInformationMessage(
+                    localize(
+                        'AWS.message.info.schemas.downloadCodeBindings.generate',
+                        'You need to be connected to AWS to select {0}.',
+                        val.label
+                    )
+                )
+
+                return undefined
```

Comment:
Just making sure, will this keep the user on the same select step, or will it back them out to the past step? If it currently backs them up a step, you may want to either:
* call recursively (or do something else) to keep the user on the same step
* display the error message in a wizard step instead of an information window that, when selected, backs the user up a step.

The main concern here is that the user's selection from the previous step isn't invalid, just this step.

Reply:
Ah I see, yes it does take the user to previous step, I will update it to show the information and the keep the user on the same step.